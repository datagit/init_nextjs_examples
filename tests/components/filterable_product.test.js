import React from 'react'
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import FilterableProductTable from '../../components/FilterableProductTable';

test('renders components FilterableProductTable', () => {
  const { getByText } = render(<FilterableProductTable />)
  const linkElement = getByText(
    /Only show products in stock/
  )
  expect(linkElement).toBeInTheDocument();
})

it('renders components FilterableProductTable to create snapshot', () => {
  const tree = renderer.create(<FilterableProductTable />).toJSON()
  expect(tree).toMatchSnapshot()
})
