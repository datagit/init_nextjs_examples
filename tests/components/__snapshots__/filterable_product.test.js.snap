// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components FilterableProductTable to create snapshot 1`] = `
<div>
  <br />
  <form>
    <p>
      <input
        aria-label="labelText"
        name="q"
        placeholder="Search..."
      />
    </p>
    <p>
      <input
        name="s"
        type="checkbox"
        value="1"
      />
       
       
      <span>
        Only show products in stock
      </span>
    </p>
  </form>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          <span
            style={Object {}}
          >
            Football
          </span>
        </td>
        <td>
          <span>
            $49.99
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={Object {}}
          >
            Baseball
          </span>
        </td>
        <td>
          <span>
            $9.99
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            Basketball
          </span>
        </td>
        <td>
          <span>
            $29.99
          </span>
        </td>
      </tr>
      <tr>
        <th
          colSpan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          <span
            style={Object {}}
          >
            iPod Touch
          </span>
        </td>
        <td>
          <span>
            $99.99
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            iPhone 5
          </span>
        </td>
        <td>
          <span>
            $399.99
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={Object {}}
          >
            Nexus 7
          </span>
        </td>
        <td>
          <span>
            $199.99
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
